src/linter/root.go

```diff
@@ -83,6 +84,34 @@ func (r *Report) CheckName() string {
 	return r.checkName
 }
 
+// MarshalJSON is used to write report in its JSON representation.
+//
+// Used for -output-json option.
+func (r *Report) MarshalJSON() ([]byte, error) {
+	type jsonReport struct {
+		CheckName string `json:"check_name"`
+		Severity  string `json:"severity"`
```

Comment:
I think that it would be simpler to provide MarshalJSON method for Severity field and make it a proper type as well.